core/src/main/kotlin/io/specmatic/conversions/APIKeyInHeaderSecurityScheme.kt

```diff
@@ -32,4 +32,8 @@ data class APIKeyInHeaderSecurityScheme(val name: String, private val apiKey:Str
     }
 
     override fun isInRow(row: Row): Boolean = row.containsField(name)
+
+    override fun isInRequest(request: HttpRequest, complete: Boolean): Boolean {
```

Comment:
maybe this could be request.containsHeader(name) so that we don't use the headers property directly.